packages/x-date-pickers-pro/src/DateRangePicker/DateRangePicker.test.tsx

```diff
@@ -10,47 +10,48 @@ import {
 import { pickerPopperClasses } from '@mui/x-date-pickers/internals/components/PickerPopper';
 
 describe('<DateRangePicker />', () => {
-  const { render, clock } = createPickerRenderer({
-    clock: 'fake',
-    clockConfig: new Date(2018, 0, 1, 0, 0, 0, 0),
-  });
+  const { render } = createPickerRenderer();
 
   const { renderWithProps } = buildFieldInteractions({
     render,
-    clock,
     Component: DateRangePicker,
   });
 
+  const originalMatchMedia = window.matchMedia;
+
+  afterEach(() => {
+    window.matchMedia = originalMatchMedia;
+  });
+
   it('should not use the mobile picker by default', () => {
     // Test with accessible DOM structure
+    window.matchMedia = stubMatchMedia(true);
```

Comment:
Could also create a `stubMatchMedia` helper that cleans up after itself automatically

```tsx
import { onTestFinished } from 'vitest'

function stubMatchMedia(matches) {
  const original = window.matchMedia
  window.matchMedia = ...
  onTestFinished(() => {
    window.matchMedia = original
  })
}
```

The hook is called in reverse order, so it should be safe to call `stubMatchMedia` multiple times.